packages/aws-cdk-lib/aws-dynamodb/lib/table.ts

```diff
@@ -44,6 +45,169 @@ export interface SchemaOptions {
+   * - comma (`,`)
+   * - tab (`\t`)
+   * - colon (`:`)
+   * - semicolon (`;`)
+   * - pipe (`|`)
+   * - space (` `)
+   *
+   * @default - use comma as a delimiter.
+   */
+  readonly delimiter?: string;
+
+  /**
+   * List of the headers used to specify a common header for all source CSV files being imported.
+   *
+   * **NOTE**: If this field is specified then the first line of each CSV file is treated as data instead of the header.
+   * If this field is not specified the the first line of each CSV file is treated as the header.
+   *
+   * @default - the first line of the CSV file is treated as the header
+   */
+  readonly headerList?: string[];
+}
+
+/**
+ * The format of the source data.
+ */
+export abstract class InputFormat {
+  /**
+   * DynamoDB JSON format.
+   */
+  public static dynamoDBJson(): InputFormat {
+    return new class extends InputFormat {
+      public _render(): Pick<CfnTable.ImportSourceSpecificationProperty, 'inputFormat' | 'inputFormatOptions'> {
+        return {
+          inputFormat: 'DYNAMODB_JSON',
+        };
+      }
+    }();
+  }
+
+  /**
+   * Amazon Ion format.
+   */
+  public static ion(): InputFormat {
+    return new class extends InputFormat {
+      public _render(): Pick<CfnTable.ImportSourceSpecificationProperty, 'inputFormat' | 'inputFormatOptions'> {
+        return {
+          inputFormat: 'ION',
+        };
+      }
+    }();
+  }
+
+  /**
+   * CSV format.
+   */
+  public static csv(options?: CsvOptions): InputFormat {
+    // We are using the .length property to check the length of the delimiter.
+    // Note that .length may not return the expected result for multi-codepoint characters like full-width characters or emojis,
+    // but such characters are not expected to be used as delimiters in this context.
+    if (options?.delimiter && (!this.validCsvDelimiters.includes(options.delimiter) || options.delimiter.length !== 1)) {
+      throw new Error([
+        'Delimiter must be a single character and one of the following:',
+        `${this.readableValidCsvDelimiters.join(', ')},`,
+        `got '${options.delimiter}'`,
+      ].join(' '));
+    }
+
+    return new class extends InputFormat {
+      public _render(): Pick<CfnTable.ImportSourceSpecificationProperty, 'inputFormat' | 'inputFormatOptions'> {
+        return {
+          inputFormat: 'CSV',
+          inputFormatOptions: {
+            csv: {
+              delimiter: options?.delimiter,
+              headerList: options?.headerList,
+            },
+          },
+        };
+      }
+    }();
+  }
+
+  /**
+   * Valid CSV delimiters.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-csv.html#cfn-dynamodb-table-csv-delimiter
+   */
+  private static validCsvDelimiters = [',', '\t', ':', ';', '|', ' '];
+
+  private static readableValidCsvDelimiters = ['comma (,)', 'tab (\\t)', 'colon (:)', 'semicolon (;)', 'pipe (|)', 'space ( )'];
```

Comment:
Thanks @paulhcsun.
This backslash is intended.
Because `\t` is an escape sequence in js, it is necessary to escape `\t` with a backslash to display it as a string.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#white_space